### PR TITLE
feat: Added APIGW V2 Authorizer property mapping

### PR DESCRIPTION
**Pull request by lucashuy**
#### Which issue(s) does this change fix?

None.

#### Why is this change necessary?
Adds the property mappings between the Terraform and Cloudformation V2 Authorizer resources.

#### How does it address the issue?
Uses the existing mapping logic to match the Terraform properties to their equivalent Cloudformation counter parts.

#### What side effects does this change have?
None.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py [hunk 1]
```diff
@@ -941,6 +943,44 @@ def setUp(self) -> None:
             "Metadata": {"SamResourceId": f"aws_apigatewayv2_integration.{self.apigwv2_integration_name}"},
         }
 
+        self.tf_apigwv2_authorizer_common_attributes: dict = {
```
Comment (mildaniel) [inline tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py:946]: We shoudl add this authorizer to the `self.tf_json_with_root_module_only` list below for the transformation to be tested.

Comment (lucashuy) [reply tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py:946]: Good catch

File: tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py [hunk 2]
```diff
@@ -1022,9 +1023,10 @@ def setUp(self) -> None:
                 f"AwsApigatewayv2RouteMyApigwv2Route{self.mock_logical_id_hash}": self.expected_cfn_apigwv2_route,
                 f"AwsApigatewayv2StageMyApigwv2Stage{self.mock_logical_id_hash}": self.expected_cfn_apigwv2_stage,
                 f"AwsApigatewayv2IntegrationMyApigwv2Integration{self.mock_logical_id_hash}": self.expected_cfn_apigwv2_integration,
+                f"AwsApigatewayv2AuthorizerMyAuthorizerV2{self.mock_logical_id_hash}": self.expected_cfn_apigwv2_authorizer,
             },
         }
-
+        self.maxDiff = None
```
Comment (mildaniel) [inline tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py:1029]: Why do we need to set this?

Comment (lucashuy) [reply tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py:1029]: Whoops, left over from debugging, will remove

**Review by mildaniel — APPROVED** [review]

**Comment by lucashuy** [comment]
Superseded by https://github.com/aws/aws-sam-cli/pull/5625

**Pull request closed without merging**